### PR TITLE
Allow users to index the line column, for complex raw queries. Add In…

### PR DESCRIPTION
**Pull request by andreweskeclarke**
…dexConfig to wrap our numerous boolean flags

### CODE REVIEW

**Comment by andreweskeclarke** [comment]
@mattgodbolt I added a config wrapper around the boolean flags, but wasn't sure if I followed normal conventions in placing it in Index, or adding the constructors.

File: src/Index.h [hunk 1]
```diff
@@ -32,6 +32,18 @@ class Index {
...
+    struct IndexConfig {
+        bool numeric = false;
+        bool unique = false;
+        bool sparse = false;
+        bool indexLineOffsets = false;
+        IndexConfig(bool n, bool u, bool s, bool i) :
+                numeric{n}, unique{u}, sparse{s}, indexLineOffsets{i} {};
+
+        IndexConfig() :
+                numeric{false}, unique{false}, sparse{false}, indexLineOffsets{false} {};
```
Comment (mattgodbolt) [inline src/Index.h]: Maybe get rid of the "= false" on 36-39, or make this empty to default them all, else we're specifying the false in two places redundantly.

Also I'd be tempted to kill lines 40 and 41: this class is handy as it names all the fields. Having a constructor that lets you pass (true,false,false,true) just opens the door to the same error as before :)

File: tests/IndexTest.cpp [hunk 2]
```diff
@@ -50,7 +51,7 @@ TEST_CASE("indexes files", "[Index]") {
                                testFile, testFile + ".zindex");
         unique_ptr<LineIndexer> indexer(new RegExpIndexer("^Line ([0-9]+)"));
         builder
-                .addIndexer("default", "blah", true, true, false, move(indexer))
+                .addIndexer("default", "blah", Index::IndexConfig{ true, true, false, false }, move(indexer))
```
Comment (mattgodbolt) [inline tests/IndexTest.cpp]: if you make the changes from above I'd suggest making some named configs to use in these tests. "Index::IndexConfig uniqueNumeric;" etc

It'd be overkill but I often use the "builder" pattern for this:

`Index::IndexConfig().wUnique(true).wNumeric(true)` or similar. Anything's good though!

**Comment by andreweskeclarke** [comment]
@mattgodbolt I went with the IndexConfig.withNumeric(true)... approach, and removed the redundant constructor and fields.

**Comment by mattgodbolt** [comment]
LGTM! thanks!

**Pull request merged**